New Workload: TypeScript standalone

camillobruni opened the pull request:
Runs a in-memory typescript compilation of the `immer` library.

- `npm run build` downloads and build `src/gen` in-memory file contents for 3 different libs (for local testing)
- webpack is used to bundle the results
- There are some type warnings, but all files seem to be loaded

kmiller68 commented on TypeScript/benchmark.js:
```diff
@@ -0,0 +1,9 @@
+
```
Do you want a copyright here?

kmiller68 reviewed (commented):
Seems like a good workload, although sadly on it's last legs... I guess it will become a wasm benchmark when the Go transition happens 🙂 

Can we add a README.md with the steps to rebuild?

camillobruni commented:
Added README and license.

camillobruni commented:
- Updated the benchmark to load the config fils (tsconfig.json and files.json) separate and feed them to the TS compiler
- Updated webpack to use the UnicodeEscapePlugin just to be sure (will double check again after landing the 2-byte test)

kmiller68 commented:
I spoke to Yusuke offline, we think the workload is fine. Maybe we can make the migration to Go slower than the JS version 🙂 and if/when that happens we can replace this with something else for JS4+

kmiller68 reviewed (approved):
Overall, LGTM but we can also address those after merging.

camillobruni commented on JetStreamDriver.js:
```diff
@@ -2041,6 +2041,24 @@ let BENCHMARKS = [
+            "tsconfig": "./TypeScript/src/gen/immer-tiny/tsconfig.json",
+            "files": "./TypeScript/src/gen/immer-tiny/files.json",
+        },
+        iterations: 3,
+        worstCaseCount: 2,
```
DONE: switched to single iteration and adapted the subScores() helper to only conditionally emit Worst and Average.
